### PR TITLE
PREPROCESS: Twitter, NLTK (no punctuation) tokenizers

### PR DESCRIPTION
**Pull request by david-novak**
… order. Added a new option to generate tokens without punctionation.

### CODE REVIEW

File: orangecontrib/text/preprocess.py [hunk 1]
```diff
@@ -57,8 +57,11 @@ def __init__(self, lowercase=True, stop_words=None,
...
         """
         # Tokenizer.
-        self.tokenizer = word_tokenize
-        if use_twitter_tokenizer:
+        if tokenizer == 'default':
+            self.tokenizer = word_tokenize
+        elif tokenizer == 'no_punct':
+            self.tokenizer = RegexpTokenizer(r'\w+').tokenize
+        elif tokenizer == 'twitter':
             self.tokenizer = None   # TODO: Change when twitter is available.
```
Comment (nikicc) [inline orangecontrib/text/preprocess.py]: Could you also add Twitter tokenizer? It is available in NLTK as [nltk.tokenize.casual.TweetTokenizer](http://www.nltk.org/api/nltk.tokenize.html#module-nltk.tokenize.casual). Not sure how we missed this before 🙈

Comment (david-novak) [inline orangecontrib/text/preprocess.py]: Will do. The only thing is that now the tokenizers are referenced as strings when building a preprocessor. But I guess that is fine until we agree on a better sollution.

File: orangecontrib/text/widgets/owpreprocess.py [hunk 2]
```diff
@@ -548,8 +553,8 @@ def export_data(self):
 
 PREPROCESSOR_MODULES = [
     TokenizerModule,
-    TransformationModule,
     CasingModule,
+    TransformationModule,
```
Comment (nikicc) [inline orangecontrib/text/widgets/owpreprocess.py]: Why this change? `TransformationModule` should be before `CasingModule` as is the case in GUI. This will reintroduce a bug (since `persistent_data` will not be in sync with `PREPROCESSOR_MODULES`) that was fixed [a few days ago](https://github.com/biolab/orange3-text/commit/f6d765dbe166d01a59dcf160d7c45a0b74a2a627).

**Comment by nikicc** [comment]
Please, add some tests so codecov will not fail any more 😄

File: orangecontrib/text/widgets/owpreprocess.py (hunk 2 above)
Comment (david-novak) [inline orangecontrib/text/widgets/owpreprocess.py]: Nevermind, now I know why changing the order actually reintroduced the bug after rebasing to master hah. Will discard this change.

**Comment by david-novak** [comment]
@nikicc Updated the tests, should be ok now. I removed the WiP tag, but if there's anything else, just let me know.

**Comment by nikicc** [comment]
@david-novak since the second commit just reverts changes done in the first one, could you squash it into the first one? Also then you can shorten the message of the first commit to just `Added a new option to generate tokens without punctionation.`

**Comment by david-novak** [comment]
@nikicc Sure. Could also squash them all into a single commit if that's preferable?

**Comment by nikicc** [comment]
> Could also squash them all into a single commit if that's preferable?

Maybe you could also squash the last one into the third one. But otherwise I would leave two separate commits so the messages will not be too long. Also they work on two different things so I think it is OK to leave them separated.

**Comment by david-novak** [comment]
@nikicc Rebasing done. Pushed to branch.
